src/object_async/hello_async.cpp

```diff
@@ -165,8 +168,11 @@ struct AsyncHelloWorker : Nan::AsyncWorker
 {
 
     using Base = Nan::AsyncWorker;
+    // Make this class noncopyable
+    AsyncHelloWorker(AsyncHelloWorker const&) = delete;
+    AsyncHelloWorker& operator=(AsyncHelloWorker const&) = delete;
 
-    AsyncHelloWorker(bool louder, std::string const& name,
+    AsyncHelloWorker(bool louder, const std::string* name,
```

Comment:
Why by const ptr? Storing a const ref as a member and passing a const ref here does not copy the string. The difference is now the ptr can be null - with references it can not.

Reply:
The idea currently is that this is safe because the calling class `HelloObjectAsync` owns the `data_` https://github.com/mapbox/node-cpp-skel/blob/e22faa9534e0da61eb3fb5577bbb2976a3a8b09d/src/object_async/hello_async.cpp#L45-L46.  `HelloObjectAsync` will be alive for the whole time its data is used in the threadpool because it the function in JS that is triggering the threadpool usage is a member of `HelloObjectAsync`, the `helloAsync` function.

After this PR, when we move to https://github.com/mapbox/node-cpp-skel/issues/67, then the memory will be owned by the `node::Buffer` and made `Persistant` to ensure it is kept alive.